packages/gestalt/src/contexts/OnNavigation.js

```diff
@@ -0,0 +1,54 @@
+// @flow strict
+import React, { useContext, createContext, type Context, type Element, type Node } from 'react';
+import PropTypes from 'prop-types';
+
+type EventHandlerType = ({|
+  +event: SyntheticEvent<>,
+|}) => void;
+export type onNavigationOptionsType = {|
+  [string]: Node | EventHandlerType,
+|};
+
+export const onNavigationOptionsPropType: React$PropType$Primitive<{ +[string]: mixed }> =
+  PropTypes.object;
+export type onNavigationType = ({|
+  href: string,
+  onNavigationOptions?: onNavigationOptionsType,
+|}) => ?EventHandlerType;
+
+type OnNavigationType = {| onNavigation: onNavigationType |};
+
+type Props = {|
+  children: Node,
+  onNavigation?: onNavigationType,
+|};
+
+const OnNavigation: Context<OnNavigationType | void> = createContext<OnNavigationType | void>();
+const { Provider } = OnNavigation;
+
+function OnNavigationProvider({ onNavigation, children }: Props): Element<typeof Provider> {
+  return <Provider value={onNavigation ? { onNavigation } : undefined}>{children}</Provider>;
+}
+
+function useOnNavigation({
+  href,
+  onNavigationOptions,
+}: {|
+  href: string,
+  onNavigationOptions?: onNavigationOptionsType,
+|}): * {
```

Comment:
I think we know enough about the return type to type this more specifically than `*`?

Reply:
This was a temp fix. FIXED